tools/pre-commit-githook

```diff
@@ -73,13 +73,122 @@ not_pruned () {
 }
 
 # Now run check-typo over all the files in the index
-ERRORS=0
+STATUS=0
 export OCAML_CT_PREFIX=:
 export OCAML_CT_CAT="git cat-file --textconv"
 export OCAML_CT_CA_FLAG=--cached
-git diff --diff-filter=d --staged --name-only | (while IFS= read -r path
+while IFS= read -r path
 do
   if not_pruned "$path" && ! tools/check-typo "./$path" ; then
-    ERRORS=1
+    STATUS=1
   fi
-done; exit $ERRORS)
+done < <(git diff --diff-filter=d --staged --name-only)
+
+# If any files affecting the generation of configure have been updated, test
+# whether the index includes an up-to-date configure script.
+# See also tools/ci/actions/check-configure.sh
+
+AUTOCONF_FILES=\
+'configure configure.ac VERSION aclocal.m4 build-aux/* '\
+'tools/autogen tools/git-dev-options.sh'
+
+# Convert $AUTOCONF_FILES to a BRE
+PATHS="${AUTOCONF_FILES//./\\.}"
+PATHS="${PATHS//\*/.*}"
+PATHS="${PATHS// /\\|}"
```

Comment:
I'm not fond of this hackery -- the script is already complex enough. Two proposals:
- write a correct regexp by hand
- use `grep -F` (if that is portable enough?), and use a newline-separated list for AUTOCONF_FILES
  (not sure how `build-aux/*` would have to be adapted then, probably just `build-aux/`)

If I understand correctly, the regexp is currently a bit imprecise, it would match say `tools/configure.ml`, This could be solved by more sophistication, the option `-x` if it is portable enough, or manually using `^...$`.

Reply:
The missing `-x` below is a very good point, thanks, and corrected. Aside from that, I think the regular expression is precise, modulo us actively choosing to add very strangely named files to our list?

`AUTOCONF_FILES` is used directly later in the script, so we either have to give the dictionary or the BRE. Specifying both - and requiring additional files to be correctly to both a BRE and a list - looks considerably riskier than a transformation we already do in the dependency generation (see `stdlib/Makefile`).